pyproject.toml

```diff
@@ -12,20 +12,16 @@ readme = "README.md"
 requires-python = ">=3.8"
 license = {text = "Apache 2.0"}
 dependencies = [
-    "h5py>=3.7.0",
-    "mpi4py>=3.1.4",
-    "netcdf4>=1.6.3",
-    "certifi>=2023.7.22",
-    "numpy>=1.22.4",
-    "nvidia_dali_cuda110>=1.16.0",
-    "pytest>=6.0.0",
-    "ruamel.yaml>=0.17.22",
-    "setuptools>=67.6.0",
     "torch>=1.12",
+    "numpy>=1.22.4,<1.25",
+    "h5py>=3.7.0",
     "xarray>=2023.1.0",
     "zarr>=2.14.2",
+    "nvidia_dali_cuda110>=1.16.0",
     "s3fs>=2023.5.0",
-    "scikit-learn>=1.0.2"
+    "setuptools>=67.6.0",
+    "certifi>=2023.7.22",
+    "pytest>=6.0.0",
```

Comment:
Can we create a test group? Checkout pandas, they have an all, but also a bunch of sub catagories.

https://github.com/pandas-dev/pandas/blob/main/pyproject.toml#L65

Reply:
Similarly scipy also has a dev section which would be great to add things like black, interrogate, etc.

https://github.com/scipy/scipy/blob/main/pyproject.toml#L74

Reply:
I am hesitant to add too many subgroups just yet. Won't the pre-commit install required dependencies automatically?

Reply:
I agree, getting too many is just kinda pointless... nobody uses `pip install panadas [s3]` they just do `pip install pandas s3fs`.

However I think there is merit to adding a [dev] with flake8 and black hard coded to a specific version. Akshay had some comments on this that sometimes people wont be aware of pre-commit and what its doing.

Reply:
Added this. Please refer to my comment here for usage/test: https://github.com/NVIDIA/modulus/pull/150#discussion_r1330658419

Reply:
Xarray has a nice set of options with a slick complete install:
https://github.com/pydata/xarray/blob/main/pyproject.toml#L40